experimental/traffic-portal/src/app/core/origins/table/origins-table.component.ts

```diff
@@ -0,0 +1,193 @@
+ */
+
+import { Component, type OnInit } from "@angular/core";
+import { FormControl } from "@angular/forms";
+import { MatDialog } from "@angular/material/dialog";
+import { ActivatedRoute } from "@angular/router";
+import { ColDef } from "ag-grid-community";
+import { BehaviorSubject } from "rxjs";
+import type { RequestOriginResponse } from "trafficops-types";
+
+import { OriginService } from "src/app/api";
+import { CurrentUserService } from "src/app/shared/current-user/current-user.service";
+import { DecisionDialogComponent } from "src/app/shared/dialogs/decision-dialog/decision-dialog.component";
+import type {
+	ContextMenuActionEvent,
+	ContextMenuItem,
+	DoubleClickLink,
+} from "src/app/shared/generic-table/generic-table.component";
+import { NavigationService } from "src/app/shared/navigation/navigation.service";
+
+/**
+ * OriginsTableComponent is the controller for the "Origins" table.
+ */
+@Component({
+	selector: "tp-origins",
+	styleUrls: ["./origins-table.component.scss"],
+	templateUrl: "./origins-table.component.html",
+})
+export class OriginsTableComponent implements OnInit {
+	/** List of origins */
+	public origins: Promise<Array<RequestOriginResponse>>;
+
+	/** Definitions of the table's columns according to the ag-grid API */
+	public columnDefs: ColDef[] = [
+		{
+			field: "name",
+			headerName: "Name",
+		},
+		{
+			field: "tenant",
+			headerName: "Tenant",
+		},
+		{
+			field: "isPrimary",
+			filter: "tpBooleanFilter",
+			headerName: "Primary",
+		},
+		{
+			field: "deliveryService",
+			headerName: "Delivery Service",
+		},
+		{
+			field: "fqdn",
+			headerName: "FQDN",
+		},
+		{
+			cellRenderer: "sshCellRenderer",
+			field: "ipAddress",
+			headerName: "IPv4 Address",
+		},
+		{
+			cellRenderer: "sshCellRenderer",
+			field: "ip6Address",
+			headerName: "IPv6 Address",
+		},
+		{
+			field: "protocol",
+			headerName: "Protocol",
+		},
+		{
+			field: "port",
+			headerName: "Port",
+		},
+		{
+			field: "coordinate",
+			headerName: "Coordinate",
+		},
+		{
+			field: "cachegroup",
+			headerName: "CacheGroup",
+		},
+		{
+			field: "profile",
+			headerName: "Profile",
+		},
+		{
+			field: "lastUpdated",
+			filter: "agDateColumnFilter",
+			headerName: "Last Updated",
+			hide: true,
+		},
+	];
+
+	/** Definitions for the context menu items (which act on augmented origin data). */
+	public contextMenuItems: Array<ContextMenuItem<RequestOriginResponse>> = [
+		{
+			href: (origin: RequestOriginResponse): string => `${origin.id}`,
+			name: "Edit",
+		},
+		{
+			href: (origin: RequestOriginResponse): string => `${origin.id}`,
+			name: "Open in New Tab",
+			newTab: true,
+		},
+		{
+			action: "delete",
+			multiRow: false,
+			name: "Delete",
+		},
+	];
+
+	/** Defines what the table should do when a row is double-clicked. */
+	public doubleClickLink: DoubleClickLink<RequestOriginResponse> = {
+		href: (row: RequestOriginResponse): string => `/core/origins/${row.id}`,
```

Comment:
Though it seems pedantic and trivial, testing this object's `href` method is the only thing preventing this file from having 100% coverage.